services/blog_post_processor.py

```
@@ -0,0 +1,244 @@
+            return None
+        
+        try:
+            # Extract clip ID from URL path (e.g., /clip_id -> clip_id)
+            clip_id = url.split('/')[-1]
+            
+            # Basic validation of clip ID format
+            if not clip_id or len(clip_id) < 3:
+                self.logger.warning(f"Invalid clip ID extracted from URL: {url}")
+                return None
+            
+            return clip_id
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)